shell/platform/glfw/flutter_glfw.cc

```diff
@@ -116,7 +116,7 @@ struct FlutterDesktopEngineState {
   std::unique_ptr<flutter::EventLoop> event_loop;
 
   // The plugin messenger handle given to API clients.
-  std::unique_ptr<FlutterDesktopMessenger> messenger;
+  std::shared_ptr<FlutterDesktopMessenger> messenger;
```

Comment:
Why is this changing?

Reply:
Here's an idea, if you think using the `shared_ptr` is confusing I could use a typedef'd unique_ptr.  Or maybe the `fml::RefPtr` instead (I'm not sure if that is available here).

```c++
typedef std::unique_ptr<FlutterDesktopMessengerRef, &FlutterDesktopMessengerRelease> 
    FlutterDesktopMessengerPtr;
```

Reply:
Replaced with a typedef'd unique_ptr.